dace/transformation/dataflow/hbm_bank_split.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+from typing import Any, Dict, Iterable, List, Tuple, Union
+
+from dace import data, dtypes, properties, registry
+from dace.sdfg import utils
+from dace.transformation import transformation
+from dace.sdfg import nodes as nd
+from dace import SDFG, SDFGState, memlet
+import functools
+
+
+@registry.autoregister_params(singlestate=True)
+@properties.make_properties
+class HbmBankSplit(transformation.Transformation):
+    """
+    A transformation that allow splitting an array and distribute it on another
+    array with one dimension more, or vice versa. Works with arbitrary arrays,
+    but its intended use case is to distribute data on many HBM-banks.
+    Matches any 2 AccessNodes connected by an edge, if the dimensionality of the two accessed
+    arrays differ by exactly one. The sizes of the arrays have to be large enough with
+    respect to the split executed, but this is not verified. While it is allowed to use symbolics 
+    for the shapes of the array, it is expected that each dimension is divisible by the number
+    of splits specified.
+
+    When appling an unrolled map is generated around the accessnodes, which copies the parts of 
+    the array to the target array.
+
+    Examples:
+    Distribute: Suppose for example we copy from A to B, where A has shape [100, 100] and B shape
+    [10, 100, 10]. We can distribute A in that case to B using the transformation by setting
+    split_array_info=[1, 10]. A will then be divided along it's second dimension into 10 parts
+    of size [100, 10] and distributed on B.
+    Gather: Suppose A has shape [4, 50, 50] and B has shape [100, 100]. If one sets
+    split_array_info to [2, 2] and applies the transformation, it will split 
+    equally in all dimensions.
+    Therefore A[0] will be copied to B[0:50, 0:50], A[1] to B[0:50, 50:100], A[2] to B[50:100, 0:50] and
+    A[3] to B[50:100, 50:100].
+
+    Note that simply reversing the AccessNodes for the arrays in the above examples would
+    have lead to the inverse operation, i.e. the gather would become a distribute and 
+    the other way around.
+    """
+
+    _src_node = nd.AccessNode("")
+    _dst_node = nd.AccessNode("")
+
+    # dtype=List[int]
+    split_array_info = properties.Property(
+        dtype=List,
+        default=None,
+        allow_none=True,
+        desc="Describes how many times this array is split in each dimension, "
+        "where the k-th number describes how many times dimension k is split. "
+        "If the k-th number is 1 this means that the array is not split in "
+        "the k-th dimension at all. "
+        "If None, then the transform will split the first dimension."
```

Comment:
if None, the arrey will be splitted in as many parts as `shape[0]`?

Reply:
Exactly. I will update the docstring.